python/ql/src/Exceptions/IncorrectExceptOrder.ql

```diff
@@ -14,22 +14,95 @@
  */
 
 import python
+import semmle.python.dataflow.new.internal.DataFlowDispatch
+import semmle.python.ApiGraphs
+import semmle.python.frameworks.data.internal.ApiGraphModels
 
-predicate incorrect_except_order(ExceptStmt ex1, ClassValue cls1, ExceptStmt ex2, ClassValue cls2) {
+predicate builtinException(string name) {
+  typeModel("builtins.BaseException~Subclass", "builtins." + name, "")
+}
+
+predicate builtinExceptionSubclass(string base, string sub) {
+  typeModel("builtins." + base + "~Subclass", "builtins." + sub, "")
+}
+
+newtype TExceptType =
+  TClass(Class c) or
+  TBuiltin(string name) { builtinException(name) }
+
+class ExceptType extends TExceptType {
```

Comment:
The many uses of `<class-stuff> or <builtin-stuff>` in the various member predicates suggests that it might be better to create two separate subclasses of `ExceptType` that each override the (abstract) predicates on this type in a different way.

Reply:
Perhaps, though that approach may be a bit longer.